src/Expander.hs

```diff
@@ -119,11 +119,10 @@ expandModule thisMod src = do
     lang <- mustBeModName (view moduleLanguage src)
```

Comment:
@david-christiansen this and `makeModule` are the functions where review would be the most useful. Allowing languages to define their own `#%module` means that while that custom macro is being expanded, the expander is in a weird limbo state where we are in a new module, but `makeModule` has not yet been ran. I'm not sure what state the expander should be in during this limbo state.

Reply:
It seems fine to me to be in a state where there's no module right now. I don't think we have any actions in the macro monad that assume a current module, do we? Otherwise, this seems similar to a custom declaration macro - as it expands to the core forms that create a declaration, it doesn't exist yet, right? What makes modules special here?